File: 257-Binary-Tree-Paths/Solution.java
public class Solution {
	public List<String> binaryTreePaths(TreeNode root) {
		List<String> result;
        Stack<TreeNode> stack = new Stack<>();
        stack.push(root);
        TreeNode cur = root;
        while(cur!=null) {
            while(cur.left!=null){
                stack.push(cur.left);
                cur = cur.left
            }

        }
		return result;	
	}
}	

File: 257-Binary-Tree-Paths/Solution.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun binaryTreePaths(root: TreeNode?): List<String> {
        if(root == null) {
            return listOf()
        }
        val result = ArrayList<String>()
        
        helper(root, "", result)
        return result
    }
    
    fun helper(node: TreeNode, path: String, paths: ArrayList<String>) {
        if(node.left == null && node.right == null) {
            paths.add("$path${node.`val`}")
            return
        }
        if(node.left != null) {
            helper(node.left, "$path${node.`val`}->", paths)
           
        }
        if(node.right != null) {
            helper(node.right, "$path${node.`val`}->", paths)
           
        }
    
         
    }
}
